pkg/tests/mappings/mappings_test.bzl

```diff
@@ -592,6 +608,269 @@ def _test_pkg_mkdirs():
+    attrs = {
+        "expected_pkg_files": attr.label_list(
+            providers = [PackageFilesInfo],
+            default = [],
+        ),
+        "expected_pkg_dirs": attr.label_list(
+            providers = [PackageDirsInfo],
+            default = [],
+        ),
+        "expected_pkg_symlinks": attr.label_list(
+            providers = [PackageSymlinkInfo],
+            default = [],
+        ),
+        "verify_origins": attr.bool(
+            default = True,
+        ),
+    },
+)
+
+def _test_pkg_filegroup(name):
+    pkg_files(
+        name = "{}_pkg_files".format(name),
+        srcs = ["foo", "bar"],
+        prefix = "bin",
+        tags = ["manual"],
+    )
+
+    pkg_mkdirs(
+        name = "{}_pkg_dirs".format(name),
+        dirs = ["etc"],
+        tags = ["manual"],
+    )
+
+    pkg_mklink(
+        name = "{}_pkg_symlink".format(name),
+        src = "src",
+        dest = "dest",
+        tags = ["manual"],
+    )
+
+    pkg_filegroup(
+        name = "{}_pfg".format(name),
+        srcs = [t.format(name) for t in ["{}_pkg_files", "{}_pkg_dirs", "{}_pkg_symlink"]],
+        tags = ["manual"],
+    )
+
+    # Base case: confirm that basic data translation is working
+    pkg_filegroup_contents_test(
+        name = "{}_contents_valid".format(name),
+        target_under_test = "{}_pfg".format(name),
+        expected_pkg_files = ["{}_pkg_files".format(name)],
+        expected_pkg_dirs = ["{}_pkg_dirs".format(name)],
+        expected_pkg_symlinks = ["{}_pkg_symlink".format(name)],
+    )
+
+    ##################################################
+
+    pkg_files(
+        name = "{}_pkg_files_prefixed".format(name),
+        srcs = ["foo", "bar"],
+        prefix = "prefix/bin",
+        tags = ["manual"],
+    )
+
+    pkg_mkdirs(
+        name = "{}_pkg_dirs_prefixed".format(name),
+        dirs = ["prefix/etc"],
+        tags = ["manual"],
+    )
+
+    pkg_mklink(
+        name = "{}_pkg_symlink_prefixed".format(name),
+        src = "src",
+        dest = "prefix/dest",
+        tags = ["manual"],
+    )
+
+    # Test that prefixing works by using the unprefixed mapping rules we
+    # initially created, and set a prefix.
+    pkg_filegroup(
+        name = "{}_prefixed_pfg".format(name),
+        srcs = [t.format(name) for t in ["{}_pkg_files", "{}_pkg_dirs", "{}_pkg_symlink"]],
+        prefix = "prefix",
+        tags = ["manual"],
+    )
+
+    # Then test that the results are equivalent to manually specifying the
+    # prefix everywhere.
+    pkg_filegroup_contents_test(
+        name = "{}_contents_prefix_translated".format(name),
+        target_under_test = "{}_prefixed_pfg".format(name),
+        expected_pkg_files = ["{}_pkg_files_prefixed".format(name)],
+        expected_pkg_dirs = ["{}_pkg_dirs_prefixed".format(name)],
+        expected_pkg_symlinks = ["{}_pkg_symlink_prefixed".format(name)],
+        # The origins for everythin will be wrong here, since they're derived
```

Comment:
everything

Reply:
Yup, that's a typo.  Wasn't sure to what you were referring there earlier; thanks for the in-meeting clarification.